src/main/java/io/cdap/plugin/gcp/publisher/source/SerializableFunction1.java

```diff
@@ -0,0 +1,29 @@
+/*
+ * Copyright © 2023 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.gcp.publisher.source;
+
+import scala.Function1;
+
+import java.io.Serializable;
+
+/**
+ * Interface used to mark a {@link Function1} serializable.
+ *
+ * @param <T> the type of the input to the function
+ * @param <R> the type of the result of the function
+ */
+public interface SerializableFunction1<T, R> extends Function1<T, R>, Serializable {
```

Comment:
Note: If you'll use the class instead of Lambda you won't need this, class can directly implement both interfaces

Reply:
Removed it . Converted to class instead of lambda. Will make the additional implementation changes in the next PR.